homeassistant/components/shelly/climate.py

```diff
@@ -378,6 +378,9 @@ def _handle_coordinator_update(self) -> None:
                     ]["schedule_profile_names"],
                 ]
             except InvalidAuthError:
-                self.coordinator.entry.async_start_reauth(self.hass)
+                self.hass.async_create_background_task(
+                    self.coordinator.entry.async_init_reauth(self.hass),
+                    f"reauth flow {DOMAIN} {self.coordinator.entry.entry_id}",
+                )
```

Comment:
This is missing the comment about the race?

Reply:
I haven't finished tracing this one back yet to know if we need a note to refactor later or not yet